.github/workflows/uploadFirrtlReleaseArtifacts.yml

```diff
@@ -36,45 +36,52 @@ jobs:
           files: circt-full-sources.tar.gz*
           repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+  # This job determines what the build matrix should be.  If this is a
+  # "scheduled" or "workflow dispatch" job, then only build Linux images.  If it
+  # is anything else (i.e., release) then build everything.
+  choose-matrix:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Add Linux
+        id: add-linux
+        run: echo 'out={"os":"linux","runner":"ubuntu-20.04","arch":"x64","tar":"tar czf","archive":"tar.gz","sha256":"shasum -a 256","cont":"\\","setup":"","cmake-args":"-DCMAKE_C_COMPILER=clang-12 -DCMAKE_CXX_COMPILER=clang++-12"}' >> $GITHUB_OUTPUT
+      - name: Add macOS
+        id: add-macos
+        run: echo 'out={"os":"macos","runner":"macos-11","arch":"x64","tar":"gtar czf","archive":"tar.gz","sha256":"shasum -a 256","cont":"\\","setup":"","cmake-args":""}' >> $GITHUB_OUTPUT
+      - name: Add Windows
+        id: add-windows
+        run: echo 'out={"os":"windows","runner":"windows-2019","arch":"x64","tar":"tar czf","archive":"zip","sha256":"sha256sum","cont":"`","setup":"./utils/find-vs.ps1","cmake-args":""}' >> $GITHUB_OUTPUT
```

Comment:
I would consider putting these JSON snippets into their own files (eg. strawman proposal, put them each in `.github/workflows/release-artifacts/{linux.json, macos.json, windows.json}`) and then loading those JSONs here, just because it would be easier as a human to read or understand them. I will grant that having to look in other files can be annoying so if possible to format them nicer here that's also an option.

This is just a suggestion though, it's not that important so feel free to ignore.

Reply:
I can put this in the `env` of this step. GitHub actions syntax will resolve the `env`. That should make this way more readable while avoiding needing to know about another file.

Reply:
I'd be curious to see what that looks like. Reiterating that making this nicer would be nice but is not that important so don't let this suggestion sandbag this PR.